src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -854,7 +854,7 @@ public async Task SendPingToExternalHostWithLowTtlTest()
             options.Ttl = 1;
             // This should always fail unless host is one IP hop away.
             pingReply = await ping.SendPingAsync(host, TestSettings.PingTimeout, TestSettings.PayloadAsBytesShort, options);
-            Assert.Equal(IPStatus.TimeExceeded, pingReply.Status);
+            Assert.True(pingReply.Status == IPStatus.TimeExceeded || pingReply.Status == IPStatus.TtlExpired);
```

Comment:
This generally looks OK to me but I'm not sure why you would see different status @danmoseley.
I assume you do the run on Windows? 
Maybe @filipnavara has some ideas.

Reply:
I don't know anything about ping, but it's setting TTL to the lowest possible value while timeout I believe is 20 sec. This is Windows 10.

Reply:
I'll recheck the ping code, TtlExpired makes more sense to me. (Not sure what would happen in non-priviledged case on Linux though)

Reply:
So, on non-Windows the error code in raw-socket code path is translated incorrectly:
- https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/IcmpV4MessageConstants.cs#L80
- https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/IcmpV6MessageConstants.cs#L79-L84

In the privileged code path running the `ping` command line utility it will presumably return `TimeExceeded` too but without the address and fail the test completely.

I think this should be addressed separately in a different PR.